drivers/modem/modem_context.c

```diff
@@ -0,0 +1,185 @@
+/** @file
+ * @brief Modem context helper driver
+ *
+ * A modem context driver allowing application to handle all
+ * aspects of received protocol data.
+ */
+
+/*
+ * Copyright (c) 2019 Foundries.io
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(modem_context, CONFIG_MODEM_LOG_LEVEL);
+
+#include <kernel.h>
+
+#include "modem_context.h"
+
+static struct modem_context *contexts[CONFIG_MODEM_CONTEXT_MAX_NUM];
```

Comment:
You have array of pointers here, where is the actual context allocated, did not spot it immediately.

Reply:
Users allocate their own `modem_context` like this in the ublox sara driver:
https://github.com/zephyrproject-rtos/zephyr/pull/18103/files#diff-cb3c7d77935fa5df6ccb74a1c5e050baR149

Then call `modem_context_register()` to store the modem_context in the master list exposing it to "externals" such as modem_shell:
https://github.com/zephyrproject-rtos/zephyr/pull/18103/files#diff-cb3c7d77935fa5df6ccb74a1c5e050baR1274

This is the `modem_context_register()` function:
https://github.com/zephyrproject-rtos/zephyr/pull/18103/files#diff-395ec129e305de98c66d00aef6c64373R165

Which calls an internal modem_context_get() function to assigns the modem_context pointer to the internal list of pointers:
https://github.com/zephyrproject-rtos/zephyr/pull/18103/files#diff-395ec129e305de98c66d00aef6c64373R142

Reply:
Ok, thanks for clarification.